packages/ember-glimmer/tests/integration/components/curly-components-test.js

```diff
@@ -917,4 +921,977 @@ moduleFor('Components test: curly components', class extends RenderingTest {
     this.assertElement(this.firstChild.childNodes[2], { tagName: 'b', content: 'bold' });
   }
 
+  ['@test lookup of component takes priority over property']() {
+    this.registerComponent('some-component', {
+      template: 'some-component'
+    });
+
+    this.render(
+      '{{some-prop}} {{some-component}}',
+      {
+        'some-component': 'not-some-component',
+        'some-prop': 'some-prop'
+      }
+    );
+
+    this.assertText('some-prop some-component');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('some-prop some-component');
+  }
+
+  ['@test component without dash is not looked up']() {
+    this.registerComponent('somecomponent', {
+      template: 'somecomponent'
+    });
+
+    this.render(
+      '{{somecomponent}}',
+      {
+        'somecomponent': 'notsomecomponent'
+      }
+    );
+
+    this.assertText('notsomecomponent');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('notsomecomponent');
+
+    this.runTask(() => this.context.set('somecomponent', 'not not notsomecomponent'));
+
+    this.assertText('not not notsomecomponent');
+
+    this.runTask(() => this.context.set('somecomponent', 'notsomecomponent'));
+
+    this.assertText('notsomecomponent');
+  }
+
+  ['@htmlbars GlimmerComponent cannot be invoked with curly braces']() {
+    this.registerComponent('non-block', {
+      ComponentClass: GlimmerComponent.extend(),
+      template: 'In layout'
+    });
+
+    expectAssertion(() => {
+      this.render('{{non-block}}');
+    }, /cannot invoke the 'non-block' component with curly braces/);
+  }
+
+  ['@test non-block with properties on attrs']() {
+    this.registerComponent('non-block', {
+      template: 'In layout - someProp: {{attrs.someProp}}'
+    });
+
+    this.render('{{non-block someProp="something here"}}');
+
+    this.assertText('In layout - someProp: something here');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('In layout - someProp: something here');
+  }
+
+  ['@test non-block with properties on attrs and component class']() {
```

Comment:
ditto re: making someProp dynamic

Also, this test seems a bit silly :P I wonder if the original intention was to test that properties on the class can be looked up, something like:

```
this.registerComponent('non-block', {
  ComponentClass: Component.extend({ selfProp: 'something else' }),
  template: 'In layout - someProp: {{attrs.someProp}} - selfProp: {{selfProp}}'
});
```

If that was not already tested elsewhere, it seems good to test that here. (You'll probably want to override `init` to save off the instance, so you can mutate `selfProp` in the INUR.)